services/ark-api/ark-api/src/ark_api/utils/memory_client.py

```diff
@@ -0,0 +1,103 @@
+"""Shared memory service client utilities."""
+import logging
+from typing import Optional, List, Dict, Any
+import httpx
+from fastapi import HTTPException
+
+logger = logging.getLogger(__name__)
+
+
+def get_memory_service_address(memory_dict: Dict[str, Any]) -> str:
+    """
+    Get the memory service address from a memory resource.
+    
+    Args:
+        memory_dict: Memory resource dictionary
+        
+    Returns:
+        Service URL string
+        
+    Raises:
+        HTTPException: If memory service is not ready or has no address
+    """
+    status = memory_dict.get("status", {})
+    service_url = status.get("lastResolvedAddress")
+    
+    if not service_url:
```

Comment:
we should check status phase too, if not ready, we should return 503, otherwie if error we might return something else, anyways, returning the status phase in the exception would be helpful